fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -389,45 +388,37 @@ public void removeOldTaskInfo() {
         long currentTimeMs = System.currentTimeMillis();
 
         List<Task> currentTask = showTasks(null);
+        Iterator<Task> iterator = currentTask.iterator();
         List<Long> taskIdToDelete = Lists.newArrayList();
-        currentTask.sort((o1, o2) -> Long.signum(o1.getCreateTime() - o2.getCreateTime()));
-        int labelKeepMaxSecond = Config.label_keep_max_second;
-        int numTaskToRemove = currentTask.size() - Config.label_keep_max_num;
-        for (Task task : currentTask) {
-            if ((currentTimeMs - task.getCreateTime()) / 1000 > labelKeepMaxSecond ||
-                    numTaskToRemove > 0) {
+        while (iterator.hasNext()) {
+            Task task = iterator.next();
```

Comment:
any concurrent control?

Reply:
Here is just a count of which should be deleted, the real deletion is in dropTasks and concurrent control in it.

Reply:
I thick there will be read/write conflict if you do not get the read lock here.